hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/account/AccountAccessorImpl.java

```diff
@@ -20,26 +20,34 @@
  * ‍
  */
 
+import static com.hedera.node.app.service.evm.accounts.HederaEvmContractAliases.isMirror;
+
 import com.google.protobuf.ByteString;
 import javax.inject.Named;
 import lombok.RequiredArgsConstructor;
 import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.datatypes.Address;
 
 import com.hedera.mirror.web3.evm.store.contract.MirrorEntityAccess;
+import com.hedera.mirror.web3.repository.EntityRepository;
 import com.hedera.node.app.service.evm.accounts.AccountAccessor;
 
 @Named
 @RequiredArgsConstructor
 public class AccountAccessorImpl implements AccountAccessor {
     public static final int EVM_ADDRESS_SIZE = 20;
     private final MirrorEntityAccess mirrorEntityAccess;
+    private final EntityRepository entityRepository;
 
     @Override
     public Address canonicalAddress(Address addressOrAlias) {
-        if (mirrorEntityAccess.isUsable(addressOrAlias)) {
-            return addressOrAlias;
+        if(!isMirror(addressOrAlias.toArray())) {
+            final var entityFoundByAlias = entityRepository.findByEvmAddressAndDeletedIsFalse(addressOrAlias.toArray());
```

Comment:
Avoid the extra allocation.

```suggestion
        var addressBytes = addressOrAlias.toArrayUnsafe();
        if (!isMirror(addressBytes)) {
            final var entityFoundByAlias = entityRepository.findByEvmAddressAndDeletedIsFalse(addressBytes);
```